docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,137 @@
+---
+title: fips-with-multiple-rhel-versions
+authors:
+  - "@carbonin"
+creation-date: 2024-05-07
+last-updated: 2024-05-08
+---
+
+# Support FIPS for installers built for different RHEL releases
+
+## Summary
+
+In order for an OpenShift cluster to be considered FIPS compliant the installer
+must be run on a system with FIPS compliant openssl libraries. This means using
+a dynamically linked `openshift-install` binary against the openssl libraries
+present on our container image. Today this is not a problem because all
+`openshift-install` binaries in use have been expecting to link to RHEL 8 based
+openssl libraries, but now OpenShift 4.16 will ship and installer that requires
+RHEL 9 libraries.
+
+This will require assisted-service to maintain a way to run the
+`openshift-install` binary in a compatible environment for multiple openssl
+versions.
+
+## Motivation
+
+FIPS compliance is important for our customers and assisted-service should be
+able to install FIPS compliant clusters.
+
+### Goals
+
+- Allow for a single installation of assisted-service to install FIPS-compliant
+  clusters using `openshift-install` binaries built against RHEL 8 or RHEL 9
+
+- Allow for FIPS compliant clusters to be installed from the SaaS offering or
+  the on-prem offering
+
+### Non-Goals
+
+- Changing cluster install interfaces to accommodate new FIPS requirements
+  should be avoided
+
+- Dynamically determining a given release's RHEL version. The RHEL version will
+  be a part of configuration tied to the minor release
+
+## Proposal
+
+Two additional containers will run alongside assisted-service in the same pod.
+These "installer-runner" containers will, when triggered by assisted-service,
```

Comment:
Are these containers expected to always run? Like even during non-fips mode?

Reply:
That's the idea, yeah.
My intention is to use these containers for both fips and non-fips cases as well since there shouldn't really be a difference and if we were to only use them for fips they would get tested very infrequently.